src/components/UNSTABLE__Pagination/PageSelector.js

```diff
@@ -37,7 +41,7 @@ function PageSelector({
     <Select
       className={classnames(namespace, className)}
       hideLabel
-      id={`${namespace}__input-${id}`}
+      id={instanceId || id}
```

Comment:
Similar to `PageInput` -- it seems ideal to generate this ID ourselves since it's just used in one spot, and people can override if they want.